src/visitor/mysql.rs

```diff
@@ -32,16 +32,47 @@ impl<'a> Mysql<'a> {
     }
 
     fn visit_numeric_comparison(&mut self, left: Expression<'a>, right: Expression<'a>, sign: &str) -> visitor::Result {
+        #[cfg(feature = "json")]
+        fn json_to_quaint_value<'a>(json: serde_json::Value) -> crate::Result<Value<'a>> {
+            match json {
+                serde_json::Value::String(str) => Ok(Value::text(str)),
+                serde_json::Value::Number(number) => {
+                    if let Some(int) = number.as_i64() {
+                        Ok(Value::integer(int))
+                    } else if let Some(float) = number.as_f64() {
+                        Ok(Value::double(float))
+                    } else {
+                        unreachable!()
+                    }
+                }
+                x => {
+                    let msg = format!("Expected JSON string or number, found: {}", x);
+                    let kind = ErrorKind::conversion(msg.clone());
+
+                    let mut builder = Error::builder(kind);
+                    builder.set_original_message(msg);
+
+                    Err(builder.build())
+                }
+            }
+        }
```

Comment:
This is the actual fix. We no longer use `CAST (? AS JSON)` but instead render the values deserialized, which works both on MySQL & MariaDB